src/test/java/net/logstash/logback/appender/LogstashTcpSocketAppenderTest.java

```diff
@@ -539,18 +556,10 @@ public void testReconnectToSecondaryOnKeepAlive() throws Exception {
         appender.setKeepAliveMessage("UNIX");
         appender.setKeepAliveDuration(Duration.buildByMilliseconds(100));
         
-        // Primary accepts first connection then refuse subsequent attempts
-        doNothing()
-            .doThrow(SocketTimeoutException.class)
-            .when(socket).connect(host("localhost", 10000), anyInt());
-
-        // Secondary refuses all attempts
-        doThrow(SocketTimeoutException.class)
-            .when(socket).connect(host("localhost", 10001), anyInt());
-
-        // Throw an exception the first time the the appender attempts to write in the output stream.
+        // Throw an exception the first time the the appender attempts to write in the output
+        // stream - this will be the keep alive message sent while on the primary destination.
         // This should cause the appender to initiate the reconnect sequence.
-        doThrow(SocketException.class)
+        doThrow(IOException.class)
             .doNothing()
             .when(outputStream).write(any(byte[].class), anyInt(), anyInt());
```

Comment:
It's unclear why this test needed to change for the refactoring of the executor service.  Can you clarify?

Reply:
Not related to the feature covered by the PR (indeed). 
The initial version threw a SocketException. This may wrongly let you think that the test case is about connection related issues which is not the case. In fact, the test case is about any kind of IOException thrown when writing in the output stream. I decided to change the type of the exception thrown to suppress any ambiguity...